### PR TITLE
✨ get end-user's flags & segments

### PR DESCRIPTION
**Pull request by deleteLater**
get end-user's flags and segments

### CODE REVIEW

File: modules/back-end/src/Application/EndUsers/EndUserFlag.cs [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+{
+    public string Name { get; set; }
+
+    public string Key { get; set; }
+
+    public string VariationType { get; set; }
+
+    public string Variation { get; set; }
+
+    public int VariationDisplayOrder { get; set; }
```
Comment (cosmos-explorer) [inline modules/back-end/src/Application/EndUsers/EndUserFlag.cs]: Why this? We don't have display order any more

File: modules/back-end/src/Application/EndUsers/EndUserFlag.cs [hunk 2]
```diff
@@ -0,0 +1,28 @@
+using Domain.FeatureFlags;
```
Comment (cosmos-explorer) [inline modules/back-end/src/Application/EndUsers/EndUserFlag.cs]: Do we really need to create a class for EndUserFlag? The FeatureFlag class is not suitable?

File: modules/back-end/src/Application/EndUsers/EndUserSegment.cs [hunk 3]
```diff
@@ -0,0 +1,19 @@
+using Domain.Segments;
```
Comment (cosmos-explorer) [inline modules/back-end/src/Application/EndUsers/EndUserSegment.cs]: Is this class really needed?

File: modules/back-end/src/Application/EndUsers/GetEndUserFlags.cs [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+    {
+        _flagService = flagService;
+        _endUserService = endUserService;
+        _evaluator = evaluator;
+    }
+
+    public async Task<IEnumerable<EndUserFlag>> Handle(GetEndUserFlags request, CancellationToken cancellationToken)
+    {
+        var endUser = await _endUserService.GetAsync(request.Id);
+        var flags = await _flagService.FindManyAsync(x => !x.IsArchived);
```
Comment (cosmos-explorer) [inline modules/back-end/src/Application/EndUsers/GetEndUserFlags.cs]: You may forgot to filter by envId

File: modules/back-end/src/Application/EndUsers/GetEndUserSegments.cs [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+    public GetEndUserSegmentsHandler(IEndUserService endUserService, ISegmentService segmentService)
+    {
+        _endUserService = endUserService;
+        _segmentService = segmentService;
+    }
+
+    public async Task<IEnumerable<EndUserSegment>> Handle(GetEndUserSegments request, CancellationToken cancellationToken)
+    {
+        var endUser = await _endUserService.GetAsync(request.Id);
+        var segments = await _segmentService.FindManyAsync(x => !x.IsArchived);
```
Comment (cosmos-explorer) [inline modules/back-end/src/Application/EndUsers/GetEndUserSegments.cs]: You may forgot to filter by envId

File: modules/front-end/src/app/core/components/user-segments-flags-drawer/user-segments-flags-drawer.component.html [hunk 6]
```diff
@@ -43,21 +35,21 @@
...
-                  <i (click)="copyText($event, item.keyName)" class="copy-icon" nz-icon nzType="icons:icon-copy"></i>
-                  {{item.keyName}}
+                  <i (click)="copyText($event, item.key)" class="copy-icon" nz-icon nzType="icons:icon-copy"></i>
+                  {{item.key}}
                 </td>
                 <td class="option">
                   <div class="left">
-                    <div class="variation-tip {{'tip-' + (item.variation.displayOrder - 1) % 9}}"></div>
-                    <div class="variation-value">{{item.variation.variationValue}}</div>
+                    <div class="variation-tip {{'tip-' + (item.variationDisplayOrder - 1) % 9}}"></div>
```
Comment (cosmos-explorer) [inline modules/front-end/src/app/core/components/user-segments-flags-drawer/user-segments-flags-drawer.component.html]: you can use the index of the variation directly, don't need variationDisplayOrder, a comment has been put to the c# previously

**Review by cosmos-explorer — APPROVED** [review]

File: modules/back-end/src/Application/EndUsers/GetEndUserSegments.cs (hunk 5 above)
Comment (deleteLater) [reply modules/back-end/src/Application/EndUsers/GetEndUserSegments.cs]: resolved in https://github.com/featbit/featbit/pull/93/commits/a4d129fb3ec3e9dda8aaf4ff62bd6221700d0812

File: modules/back-end/src/Application/EndUsers/GetEndUserFlags.cs (hunk 4 above)
Comment (deleteLater) [reply modules/back-end/src/Application/EndUsers/GetEndUserFlags.cs]: resolved in https://github.com/featbit/featbit/pull/93/commits/a4d129fb3ec3e9dda8aaf4ff62bd6221700d0812

File: modules/back-end/src/Application/EndUsers/EndUserSegment.cs (hunk 3 above)
Comment (deleteLater) [reply modules/back-end/src/Application/EndUsers/EndUserSegment.cs]: it's needed, just like a segment view model. maybe i should call it `EndUserSegmentVm`

File: modules/back-end/src/Application/EndUsers/EndUserFlag.cs (hunk 2 above)
Comment (deleteLater) [reply modules/back-end/src/Application/EndUsers/EndUserFlag.cs]: maybe i should call it `EndUserFlagVm`

Comment (deleteLater) [reply modules/back-end/src/Application/EndUsers/EndUserFlag.cs]: resolved in https://github.com/featbit/featbit/pull/93/commits/984a45b28b7c26452116beacaaa78e7fe303f90e

File: modules/back-end/src/Application/EndUsers/EndUserSegment.cs (hunk 3 above)
Comment (deleteLater) [reply modules/back-end/src/Application/EndUsers/EndUserSegment.cs]: resolved in https://github.com/featbit/featbit/pull/93/commits/984a45b28b7c26452116beacaaa78e7fe303f90e

File: modules/back-end/src/Application/EndUsers/EndUserFlag.cs (hunk 1 above)
Comment (deleteLater) [reply modules/back-end/src/Application/EndUsers/EndUserFlag.cs]: i will move this logic to front-end

File: modules/front-end/src/app/core/components/user-segments-flags-drawer/user-segments-flags-drawer.component.html (hunk 6 above)
Comment (deleteLater) [reply modules/front-end/src/app/core/components/user-segments-flags-drawer/user-segments-flags-drawer.component.html]: resolved in https://github.com/featbit/featbit/pull/93/commits/ccf1da3496ea4199328c8a9b6f56e049b9d00cd2

File: modules/back-end/src/Application/EndUsers/EndUserFlag.cs (hunk 1 above)
Comment (deleteLater) [reply modules/back-end/src/Application/EndUsers/EndUserFlag.cs]: resolved in https://github.com/featbit/featbit/pull/93/commits/ccf1da3496ea4199328c8a9b6f56e049b9d00cd2

**Review by cosmos-explorer — APPROVED** [review]

**Pull request merged**